tests/recipe/test_amplitude_creation.py

```diff
@@ -15,32 +15,42 @@ def _create_helicity_recipe():
     return recipe
 
 
+def _get_particle_list(recipe):
+    particle_list = extract_particles(recipe)
+    return particle_list
+
+
+def _get_kinematics(recipe):
+    kinematics = HelicityKinematics.from_recipe(recipe)
+    return kinematics
+
+
+def _generate_phsp(recipe, number_of_events):
+    kinematics = _get_kinematics(recipe)
+    phsp_sample = generate_phsp(number_of_events, kinematics)
+    return phsp_sample
+
+
 class TestRecipeHelicity:
-    recipe = _create_helicity_recipe()
     number_of_phsp_events = 10
+    recipe = _create_helicity_recipe()
+    particle_list = _get_particle_list(recipe)
+    kinematics = _get_kinematics(recipe)
+    phsp_sample = _generate_phsp(recipe, number_of_phsp_events)
```

Comment:
This could be a case for fixtures https://docs.pytest.org/en/stable/fixture.html

Reply:
Aha. New issue for this or fix in this PR? For me the important part in this PR is the fixes for the recipe file, whether the tests are smart pytest syntax or not would slow down this PR series